src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -1035,12 +1052,32 @@ sol_http_server_new(uint16_t port)
 
     server->buf_size = SOL_HTTP_REQUEST_BUFFER_SIZE;
 
-    server->daemon = MHD_start_daemon(MHD_USE_SUSPEND_RESUME,
-        port, NULL, NULL,
-        http_server_handler, server,
-        MHD_OPTION_NOTIFY_CONNECTION, notify_connection_cb, server,
-        MHD_OPTION_NOTIFY_COMPLETED, notify_connection_finished_cb, server,
-        MHD_OPTION_END);
+    if (config->security.cert && config->security.key) {
+        struct sol_blob *cert_contents, *key_contents;
+        cert_contents = sol_cert_get_contents(config->security.cert);
+        key_contents = sol_cert_get_contents(config->security.key);
+
+        if (!(cert_contents && key_contents))
+            goto err_daemon;
+
+        server->daemon = MHD_start_daemon(MHD_USE_SUSPEND_RESUME | MHD_USE_SSL,
+            config->port, NULL, NULL,
+            http_server_handler, server,
+            MHD_OPTION_NOTIFY_CONNECTION, notify_connection_cb, server,
+            MHD_OPTION_NOTIFY_COMPLETED, notify_connection_finished_cb, server,
+            MHD_OPTION_HTTPS_MEM_KEY, (char *)key_contents->mem,
+            MHD_OPTION_HTTPS_MEM_CERT, (char *)cert_contents->mem,
+            MHD_OPTION_END);
+        sol_blob_unref(cert_contents);
+        sol_blob_unref(key_contents);
```

Comment:
Just thinking, this information is sensitive, so perhaps it is a good idea to clear this memory when are done using it, to reduce any risk of exposing it.